### PR TITLE
CI: check docs of private items

### PR DESCRIPTION
**Pull request by MarcoIeni**
## Motivation

Documentation of private items is not tested in CI.

## Solution

Adding the `--document-private-items` should check all documentation.

### CODE REVIEW

**Comment by Darksonn** [comment]
Thanks. You will need to also fix the new doc failures.

**Comment by MarcoIeni** [comment]
I fixed the "obvious" ones. I looked at the remaining warnings. They belong to docs which look outdated to me because they refer to functions that do not exist anymore.
I don't have knowledge of the internals of the tokio library (and I don't have the time to learn it unfortunately right now 😅) so if it's ok for you I would live this task to someone else by substituting all the links (i.e. `[``]`) to normal markdown code (``) and I will open two separate issues, one for each of the outdated docs.

**Comment by Darksonn** [comment]
That would be fine.

**Comment by MarcoIeni** [comment]
These are the issues I will open:

# Fix documentation of PollEvented

The [PolledEvent](https://github.com/tokio-rs/tokio/blob/1e2f893da2f33deef7fb7c6dc62b265f454c4456/tokio/src/io/poll_evented.rs#L68-L72) struct documentation needs to be reviewed.
For example, it refers to the functions `clear_write_ready` and `clear_read_ready` which do not exist.

# Fix documentation of Registration::new_with_interest_and_handle

The [new_with_interest_and_handle](https://github.com/tokio-rs/tokio/blob/1e2f893da2f33deef7fb7c6dc62b265f454c4456/tokio/src/io/driver/registration.rs#L69) method documentation needs to be reviewed.
For example, it refers to the functions `new_with_interest` and `new`, which do not exist.

File: tokio/src/util/rand.rs [hunk 1]
```diff
@@ -3,7 +3,7 @@ use std::cell::Cell;
 /// Fast random number generate
 ///
 /// Implement xorshift64+: 2 32-bit xorshift sequences added together.
-/// Shift triplet [17,7,16] was calculated as indicated in Marsaglia's
+/// Shift triplet \[17,7,16\] was calculated as indicated in Marsaglia's
```
Comment (Darksonn) [inline tokio/src/util/rand.rs:6]: I prefer this.
```suggestion
/// Shift triplet `[17,7,16]` was calculated as indicated in Marsaglia's
```

File: tokio-stream/src/stream_map.rs [hunk 2]
```diff
@@ -605,7 +605,7 @@ mod rand {
     /// Fast random number generate
     ///
     /// Implement xorshift64+: 2 32-bit xorshift sequences added together.
-    /// Shift triplet [17,7,16] was calculated as indicated in Marsaglia's
+    /// Shift triplet \[17,7,16\] was calculated as indicated in Marsaglia's
```
Comment (Darksonn) [inline tokio-stream/src/stream_map.rs:608]:
```suggestion
    /// Shift triplet `[17,7,16]` was calculated as indicated in Marsaglia's
```

**Review by Darksonn — COMMENTED** [review]
Thanks!

**Comment by MarcoIeni** [comment]
In a7a9464 I tried to fix the last error CI was giving me in 4e85d67.
By the way I couldn't reproduce it locally, so I hope it works in CI now 😅

**Review by Darksonn — APPROVED** [review]
Assuming the PR passes CI, then this is ready to be merged.

**Comment by taiki-e** [comment]
Seems CI is [still failing](https://github.com/tokio-rs/tokio/pull/3715/checks?check_run_id=2420585709). You probably have to bump the version of the nightly compiler in CI.
https://github.com/tokio-rs/tokio/blob/1e2f893da2f33deef7fb7c6dc62b265f454c4456/.github/workflows/ci.yml#L12

**Comment by MarcoIeni** [comment]
Looks like the error is still there, so I will revert the nightly bump change.

**Comment by MarcoIeni** [comment]
Out of curiosity, why are we using nightly for docs?

**Comment by Darksonn** [comment]
Because we use the unstable [`doc_cfg`][1] feature for our documentation.

[1]: https://doc.rust-lang.org/unstable-book/language-features/doc-cfg.html

**Comment by Darksonn** [comment]
Thanks for the PR!

**Pull request merged**